docs/scripts/build-props-table.mjs

```diff
@@ -0,0 +1,64 @@
+import fs from 'fs-extra';
+import path from 'path';
+import { dirname } from 'path';
+import reactDocgenTypescript from 'react-docgen-typescript';
+import { fileURLToPath } from 'url';
+
+console.log('📑 Generating props table...');
+
+const parser = reactDocgenTypescript.withCustomConfig('./tsconfig.json', {
+  shouldRemoveUndefinedFromOptional: true,
+  shouldExtractLiteralValuesFromEnum: false,
+  shouldExtractValuesFromUnion: false,
+
+  // include asserted types for components eg. Autocomplete & Select
+  customComponentTypes: ['AutocompleteComponent', 'SelectComponent'],
+});
+
+// Resolve __dirname for ESM
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const componentsDir = path.resolve(__dirname, '../../packages/components/src');
+const outputFilePath = path.resolve(__dirname, '../.table-props/index.json');
```

Comment:
I find table props somewhat confusing. Rather call it "component props"?, maybe we can put it next to the registry file? 

The file should be added to gitignore

Reply:
:heavy_check_mark: 
didn't get "we put it next to reg file "